### PR TITLE
file_system: limit thread_io target os

### LINKED ISSUE #11698: tikv build fail on darwin
## Bug Report

```
[2021-12-21T17:57:18.712Z] error[E0432]: unresolved import `nix::unistd::gettid`

[2021-12-21T17:57:18.712Z]   --> components/file_system/src/thread_io.rs:15:27

[2021-12-21T17:57:18.712Z]    |

[2021-12-21T17:57:18.712Z] 15 | use nix::unistd::{getpid, gettid, Pid};

[2021-12-21T17:57:18.712Z]    |                           ^^^^^^

[2021-12-21T17:57:18.712Z]    |                           |

[2021-12-21T17:57:18.712Z]    |                           no `gettid` in `unistd`

[2021-12-21T17:57:18.713Z]    |                           help: a similar name exists in the module: `getpid`
```

### What operating system and CPU are you using?

master

### PR DESCRIPTION
**Pull request by MuZhou233**
Signed-off-by: MuZhou233 <muzhou233@outlook.com>

### What is changed and how it works?

Close #11698 
Caused by #11667

What's Changed:
- `thread_io` only support on linux now

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/file_system/src/thread_io.rs [hunk 1]
```diff
@@ -1,187 +1,218 @@
...
-            last_bytes: IOBytes::default(),
-        }
+#[cfg(target_os = "linux")]
+pub use self::linux::{fetch_all_thread_io_bytes, get_io_type, set_io_type};
+
+#[cfg(target_os = "linux")]
+use self::linux::fetch_thread_io_bytes;
+
+#[cfg(not(target_os = "linux"))]
+mod notlinux {
```
Comment (sticnarf) [inline components/file_system/src/thread_io.rs:198]: It's not idiomatic to combine words as an identifier, so I prefer change the mod name to `non_linux`.

**Review by sticnarf — COMMENTED** [review]
Please add `close #xxx` to the commit message to pass the check :)

**Review by Connor1996 — COMMENTED** [review]
rest LGTM

**Review by Connor1996 — APPROVED** [review]
LGTM

**Review by sticnarf — APPROVED** [review]
Thanks!

**Pull request merged**